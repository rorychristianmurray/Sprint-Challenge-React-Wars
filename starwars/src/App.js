import React, { useState, useEffect } from "react";
import axios from "axios";
import CharacterList from "./CharacterList";
import "./App.css";

const App = () => {
  // Try to think through what state you'll need for this app before starting. Then build out
  // the state properties here.

  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a
  // side effect in a component, you want to think about which state and/or props it should
  // sync up with, if any.
  const [chars, setChars] = useState({});

  const fetchChars = () => {
    axios
      .get(`https://swapi.co/api/people/`)
      .then(response => {
        console.log("fetchChars response.data", response.data);
        setChars(response.data);
      })
      .catch(error => {
        console.log("fetchChars error", error);
      });
  };

  useEffect(() => {
    fetchChars();
  }, []);

  return (
    <div className="App">
      <h1 className="Header">React Wars</h1>
      <CharacterList chars={chars} />
    </div>
  );
};

export default App;
