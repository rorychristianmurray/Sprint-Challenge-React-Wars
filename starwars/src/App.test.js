import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import renderer from "react-test-renderer";

describe("<App />", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
