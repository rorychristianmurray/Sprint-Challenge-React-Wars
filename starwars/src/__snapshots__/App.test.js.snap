// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="Header"
  >
    React Wars
  </h1>
  <h3>
    Loading...
  </h3>
</div>
`;
